### PR TITLE
feat: add getters for values, nodes and attributes

### PR DESCRIPTION
**Pull request by asilenzi**
and implement some generators.

@raulcostajunior

```
  helper
    ✔ helper simple keys
    ✔ helper vector strings
    ✔ helper vector strings
    ✔ helper vector floats
    ✔ helper vector floats
    ✔ helper vector empty arrays
    ✔ helper vector sub hashes
    ✔ helper vector hash

  binary
    ✔ reading
    ✔ writing
    ✔ schema

  11 passing (36ms)
```

### CODE REVIEW

File: tests/karabo-ts-tests/serializer.ts [hunk 1]
```diff
@@ -16,16 +16,50 @@ describe("binary", function() {
...
+        expect(hsh.value_["uint8PropertyReadOnly"].value.value_).to.equal(177);
+        expect(hsh.getValue("deviceId")).to.equal("Bob");
+        expect(hsh.getValue("classId")).to.equal("PropertyTest");
+        expect(hsh.getValue("alarmCondition")).to.equal("none");
+        expect(hsh.getValue("boolPropertyReadOnly")).to.equal(false);
+        expect(hsh.getValue("uint16PropertyReadOnly")).to.equal(32000);
+        expect(hsh.getValue("int16PropertyReadOnly")).to.equal(3200);
+        expect(hsh.getValue("uint32PropertyReadOnly")).to.equal(32000000);
+        expect(hsh.getValue("floatPropertyReadOnly")).to.equal(3.1415960788726807);
+        expect(hsh.getValue("uint8PropertyReadOnly")).to.equal(177);
```
Comment (asilenzi) [inline tests/karabo-ts-tests/serializer.ts]: @raulcostajunior I can add a example of the `.` syntax

Comment (asilenzi) [reply tests/karabo-ts-tests/serializer.ts]: done

**Comment by asilenzi** [comment]
Of the added lines, 145 are tests

**Comment by raulcostajunior** [comment]
Nice tests for the helpers!

I only have a small suggestion for the exposure of `value_`. If you decide to take it, please feel free to leave it for it another PR.

LGTM

**Comment by asilenzi** [comment]
> Nice tests for the helpers!
> 
> I only have a small suggestion for the exposure of `value_`. If you decide to take it, please feel free to leave it for it another PR.
> 
> LGTM

~~Would you mind sharing the suggestion?~~

Edit: Discussed offline. We keep the `value_` as is for the moment.

I added a few return type annotations

**Comment by asilenzi** [comment]
done: Could you have a look @raulcostajunior ?

**Comment by raulcostajunior** [comment]
LGTM after reviewing of the latest commit

**Comment by asilenzi** [comment]
Ok, sorry to make you look again.

I pushed one last commit. This extends the tests including a round trip test of the hashes created by `makeHash`

I fixed the issues that would have caused to Kiwi

**Comment by raulcostajunior** [comment]
LGTM after reviewing the latest commit. Nice addition, the `roundtrip` tests!

Please ignore the suggestion to include the `Int64` type for numbers outside the `Int32` range but still within the `*_SAFE_INTEGER` range if it triggers any other change or can cause instabilities (I can't gauge the implications of such a change).

**Comment by asilenzi** [comment]
Thanks a lot for the feedback and your patience!

I think I have handled the cases you mentioned.

**Comment by raulcostajunior** [comment]
Cool!

**Pull request merged**